### ISSUE TITLE
Oxtrust allow to update user with duplicate uid when users are created through SCIM

### ISSUE THREAD

**Issue by sahiliamsso** [GluuFederation/oxTrust#1028]
Based on [this](https://support.gluu.org/identity-management/5568/duplicate-uid/) ticket, OxTrust doesn't check the unique UID when users is created through SCIM
Steps to reproduce --
1.  Create UserA and UserB with SCIM API
2. Login to UI, and update UserB and change the userid to UserA.  Click on update
Expected Result - Duplicate UID error should be displayed
Actual Result - User is saved successfully

![image](https://user-images.githubusercontent.com/21104811/42189888-fb4eac10-7e77-11e8-930a-188590cde6af.png)

**Comment by mzico** [comment]
@jgomer2001

**Comment by shekhar16** [comment]
@sahiliamsso @mzico   its not possible to add duplicate uwername in ldap because we had already an unique constraints in ldap, ldap doesnt allow to save duplicate UID.
@yurem

**Comment by shekhar16** [comment]
@sahiliamsso @mzico  Can u please share the oxtrust version u r using ?

**Comment by shekhar16** [comment]
![image](https://user-images.githubusercontent.com/15937500/42734132-57ee92c6-885c-11e8-8963-8b21cbf6fdde.png)

**Comment by syntrydy** [comment]
Fixed

**Comment by jgomer2001** [comment]
Hi @sahiliamsso

I just tried to replicate with no success, here is what I did on a centos6 3.1.4 RC1:

1. Created 2 users with usernames "UserA" and "UserB" with the great SCIM java client :) 
1. Confirmed in LDAP both were created, both in oxTrust appear too
1. Still in oxTrust, tried to edit user "UserB" by setting its username to "UserA"

and got the expected error:

![uidalreadyexists](https://user-images.githubusercontent.com/5444180/45384529-dfaa3180-b5d3-11e8-925e-18f5ab983a2d.png)


So basically the same  @natt-tester obtained already.  

Want to give it a try once more?
